thoth/storages/data/alembic/versions/b23e6145a6ab_added_a_model_haspythoninterpreter_to_.py

```diff
@@ -0,0 +1,38 @@
+"""Added a model HasPythonInterpreter to link python interpreters found in PackageExtractRun
+
+Revision ID: b23e6145a6ab
+Revises: 184e698fe2c2
+Create Date: 2019-10-11 14:35:56.253061+00:00
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = 'b23e6145a6ab'
+down_revision = '184e698fe2c2'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('has_python_interpreter',
+    sa.Column('id', sa.Integer(), autoincrement=True, nullable=False),
+    sa.Column('path', sa.String(length=256), nullable=False),
+    sa.Column('link', sa.String(length=256), nullable=True),
+    sa.Column('version', sa.String(length=256), nullable=True),
+    sa.Column('package_extract_run_id', sa.Integer(), nullable=False),
+    sa.Column('software_environment_id', sa.Integer(), nullable=False),
+    sa.ForeignKeyConstraint(['package_extract_run_id'], ['package_extract_run.id'], ondelete='CASCADE'),
+    sa.ForeignKeyConstraint(['software_environment_id'], ['software_environment.id'], ondelete='CASCADE'),
+    sa.PrimaryKeyConstraint('id', 'package_extract_run_id', 'software_environment_id')
```

Comment:
Not sure if this is the right approach - you are creating relation on 3 tables.

What about creating a table which would carry python interpreter information (path, version, link) and link to it from `package_extract_run` table? `has_python_interpreter` would be than capturing many-to-many relationship.